### PR TITLE
(JAVACLI-130) Add Absolute dates to filter get_detailed_objects

### PR DESCRIPTION
**Pull request by berkeleysquare**
For backwards-compatibility olderthan and newrthan are unchanged, providing relative date before now().

Added before and after for absolute Dates, e.g. before:Y2016.M10.D25,after:Y2016.M9.D30

### CODE REVIEW

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetDetailedObjects.java [hunk 1]
```diff
@@ -120,25 +123,47 @@ public boolean apply(@Nullable final DetailedS3Object input) {
...
             return Predicates.notNull();
         }
-        // if one is specified, default the other
-        final long newerThan = Guard.isStringNullOrEmpty(newer) ? 0L : new Date().getTime() - Utils.dateDiffToSeconds(newer) * MILLIS_PER_SECOND;
-        final long olderThan = Guard.isStringNullOrEmpty(older) ? Long.MAX_VALUE : new Date().getTime() - Utils.dateDiffToSeconds(older) * MILLIS_PER_SECOND;
+
+        long newerThan = 0L;
+        long olderThan = Long.MAX_VALUE;
+
+        if (!Guard.isStringNullOrEmpty(newer)) {
```
Comment (RachelTucker) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetDetailedObjects.java]: What happens if only one of these are set? What if all of them are set?

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/Utils.java [hunk 2]
```diff
@@ -307,6 +307,74 @@ public static long dateDiffToSeconds(final String diff) {
...
+        dateString.append(month);
+        dateString.append("-");
+        dateString.append(date);
+        dateString.append("T");
+        dateString.append(hours);
+        dateString.append(":");
+        dateString.append(mins);
+        dateString.append(":");
+        dateString.append(secs);
+        dateString.append(".000Z");
```
Comment (RachelTucker) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/Utils.java]: you can string the appends and do something like `dateString.append("-").append(month).append("-")`

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetDetailedObjects.java (hunk 1 above)
Comment (berkeleysquare) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetDetailedObjects.java]: If one is set the other range element defaults to Date(0) - Date(MAX). If both starts or both ends are set, the absolute date wins.

I emailed George T:
"Deranged use could even mix relative and absolute. This year, older than two days is after:Y2016,olderthan:d2 but if both are provided, (say after and newerthan) absolute would win."

**Comment by berkeleysquare** [comment]
Frighteningly hard to make the DateFormat.parse() method fail -- the method throws a ParseException but I could not make it fail with 2016-99-88 or "FredWilmaBetty" How badly should we protect user form themselves?

**Pull request closed without merging**